src/Microsoft.ML.TimeSeries/ExtensionsCatalog.cs

```diff
@@ -164,6 +165,24 @@ public static class TimeSeriesCatalog
         /// </format>
         /// </example>
         public static RootCause LocalizeRootCause(this AnomalyDetectionCatalog catalog, RootCauseLocalizationInput src, double beta = 0.5)
+        {
+            return LocalizeRootCauses(catalog, src, beta).Causes.FirstOrDefault();
```

Comment:
Is it the best way to localizeRootCause according to rootCauses? If so, for users who only care about the best dimension and related root cause, caclulations about other dimension's root cause is a waste. It would be better if we extract some bottom method instead of only choosing the first one

Reply:
I think it is OK if we accept the overall performance for RCA for all dimensions. Also what do you mean by "extract some bottom method"

Reply:
We will have two functions `LocalizeRootCause` and `LocalizeRootCauses`. 
Is it possible to make the functionality of the second as a parameter of the first and not have the second?

---
In reply to: [443748224](https://github.com/dotnet/machinelearning/pull/5236#discussion_r443748224) [](ancestors = 443748224)

Reply:
We could do that, but then we would have to change the return type of `LocalizeRootCause `or delete it. Would that be OK?

Reply:
We can't delete the API. I went through a number of different options (retaining same parameter but adding to the list of `RootCauseItem`s, creating an overloaded function, etc. and I think what you have done is more readable and understandable. No need to change it. 

---
In reply to: [446288968](https://github.com/dotnet/machinelearning/pull/5236#discussion_r446288968) [](ancestors = 446288968)